From 199ae473c80cc8f474e51a80ba9616e930ba469e Mon Sep 17 00:00:00 2001
From: Sugar Zhang <sugar.zhang@rock-chips.com>
Date: Tue, 10 Jul 2018 10:41:36 +0800
Subject: [PATCH] ASoC: rockchip: vad: fliter the playback for vad

This patch fliter the playback for vad, because vad only
handle capture.

Change-Id: I96ec24870adc4441081b39030056ed9361477bf3
Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
---
 sound/soc/rockchip/rockchip_vad.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/rockchip/rockchip_vad.c b/sound/soc/rockchip/rockchip_vad.c
index 291451d64629..dae05c9ddf24 100644
--- a/sound/soc/rockchip/rockchip_vad.c
+++ b/sound/soc/rockchip/rockchip_vad.c
@@ -581,6 +581,9 @@ static int rockchip_vad_enable_cpudai(struct snd_pcm_substream *substream)
 static int rockchip_vad_pcm_startup(struct snd_pcm_substream *substream,
 				    struct snd_soc_dai *dai)
 {
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
+		return 0;
+
 	vad_substream = substream;
 
 	return 0;
-- 
2.35.3

